connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTaskThread.java

```diff
@@ -50,16 +50,20 @@ public WorkerSinkTaskThread(WorkerSinkTask task, String name, Time time,
 
     @Override
     public void execute() {
-        // Try to join and start. If we're interrupted before this completes, bail.
-        if (!task.joinConsumerGroupAndStart())
-            return;
+        try {
+            // Try to join and start. If we're interrupted before this completes, bail.
+            if (!task.joinConsumerGroupAndStart())
+                return;
 
-        while (getRunning()) {
-            iteration();
-        }
+            while (getRunning()) {
+                iteration();
+            }
 
-        // Make sure any uncommitted data has committed
-        task.commitOffsets(true, -1);
+            // Make sure any uncommitted data has committed
+            task.commitOffsets(true, -1);
+        } catch (Throwable t) {
+            task.abort(t);
```

Comment:
Hmm, I'm having trouble reasoning about the shutdown in this case. In the existing cases, we'd have `WorkerSinkTask.stop()`/`WorkerSinkTask.awaitStop()` handle invoking these, which also tries to ensure this thread has exited and calls `SinkTask.stop()`. But in this case, this thread calls `WorkerSinkTask.abort()`, which only calls `SinkTask.stop()`. How would we manage a case where, e.g., work is rebalanced by the DistributedHerder, which would trigger the Worker to stop a task (from the Herder's thread)? Then we'd potentially have `SinkTask.stop()` called multiple times if it hit this `catch` block, right?

I think the responsibility of which thread invokes certain methods might be getting confused here. (Admittedly, it is tough to get this right since the responsibilities of different threads is confusing in this code.)